File: server/graph/model/post.go
package model

import "gorm.io/gorm"

type Post struct {
	gorm.Model
	ID       string `json:"id"`
	PosterID string
	Poster   *User      `json:"poster"`
	Text     string     `json:"text"`
	PhotoURL *string    `json:"photoUrl"`
	VideoURL *string    `json:"videoUrl"`
	Comments []*Comment `json:"comments"`
	Sends    []*User    `json:"sends" gorm:"many2many:post_sends;"`
	Likes    []*User    `json:"likes" gorm:"many2many:post_likes;"`
	Tags []*Tag `gorm:"many2many:comment_tags;"`
}

File: client/src/types/post.ts
import { Comment } from "./comment";
import { Tag } from "./tag";
import { User } from "./user";

export type Post = {
  id: string;
  poster: User;
  text: string;
  photoUrl: string;
  videoUrl: string;
  comments: Array<Comment>;
  sends: Array<User>;
  likes: Array<User>;
  tags: Array<Tag>;
  createdAt: string;
};
